File: 151002/Protchenko/lab1/rest-api/src/main/java/by/bsuir/restapi/model/entity/Author.java
package by.bsuir.restapi.model.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Author implements Entity<Long> {

    private Long id;
    private String login;
    private String password;
    private String firstname;
    private String lastname;

}

File: 151003/Halai/Publisher/Entity/Db/Author.cs
using Publisher.Entity.Common;
using System.ComponentModel.DataAnnotations;

namespace Publisher.Entity.Db
{
    public class Author(string login, string password, string firstName, string lastName) : AbstractEntity
    {
        [MinLength(2)]
        public string Login { get; set; } = login;
        [MinLength(8)]
        public string Password { get; set; } = password;
        [MinLength(2)]
        public string FirstName { get; set; } = firstName;
        [MinLength(2)]
        public string LastName { get; set; } = lastName;
        public ICollection<News> Newss { get; set; } = [];


        public Author(int id, string login, string password, string firstName, string lastName) :
            this(login, password, firstName, lastName)
        {
            Id = id;
        }

        public Author() : this(string.Empty, string.Empty, string.Empty, string.Empty) { }
    }
}
